src/block/inventory/SmithingTableInventory.php

```diff
@@ -25,13 +25,30 @@
 
 use pocketmine\inventory\SimpleInventory;
 use pocketmine\inventory\TemporaryInventory;
+use pocketmine\item\Item;
 use pocketmine\world\Position;
 
 final class SmithingTableInventory extends SimpleInventory implements BlockInventory, TemporaryInventory{
 	use BlockInventoryTrait;
 
+	public const SLOT_INPUT = 0;
+	public const SLOT_ADDITION = 1;
+	public const SLOT_TEMPLATE = 2;
+
 	public function __construct(Position $holder){
 		$this->holder = $holder;
 		parent::__construct(3);
 	}
+
+	public function getInput() : Item{
+		return $this->getItem(self::SLOT_INPUT);
+	}
+
+	public function getAddition() : Item{
+		return $this->getItem(self::SLOT_ADDITION);
+	}
+
+	public function getTemplate() : Item{
+		return $this->getItem(self::SLOT_TEMPLATE);
+	}
```

Comment:
Not much point adding wrapper functions for these. It's a stupid convention. Constants are enough.